webapp/src/js/services/hydrate-messages.js

```diff
@@ -0,0 +1,85 @@
+var _ = require('underscore');
+
+angular.module('inboxServices').factory('HydrateMessages',
+  function(
+    $q,
+    DB,
+    LineageModelGenerator
+  ) {
+
+    'use strict';
+    'ngInject';
+
+    var hydrateMessage = function(doc, key, date, report) {
```

Comment:
I think this would be clearer with a different name. "hydrate" usually means fetch it from the db. I would call this something like "buildMessageModel".